src/mscorlib/shared/System/IO/PathHelper.Windows.cs

```diff
@@ -24,63 +25,73 @@ internal class PathHelper
         /// <returns>Normalized path</returns>
         internal static string Normalize(string path)
         {
-            // Get the full path
-            StringBuffer fullPath = new StringBuffer(path.Length);
+            if (path.Length < PathInternal.MaxShortPath)
+            {
+                Span<char> initialBuffer = stackalloc char[PathInternal.MaxShortPath];
+                return Normalize(path, new ValueStringBuilder(initialBuffer));
```

Comment:
I would be cleaner to pass the ValueStringBuilder by ref, and also Dispose it here.